src/chains/ethereum/ethereum/src/helpers/simulations.ts

```diff
@@ -0,0 +1,582 @@
+   */
+  gas: Quantity;
+  /**
+   * Integer of the gasPrice used for each paid gas
+   */
+  gasPrice: Quantity;
+  /**
+   * Integer of the value sent with this transaction
+   */
+  value?: Quantity;
+  /**
+   * Hash of the method signature and encoded parameters. For details see Ethereum Contract ABI in the Solidity documentation
+   */
+  data?: Data;
+  block: RuntimeBlock;
+  /**
+   * Array of addresses and storage keys.
+   */
+  accessList?: AccessList;
+};
+
+type CallOverride =
+  | Partial<{
+      code: string;
+      nonce: string;
+      balance: string;
+      state: { [slot: string]: string };
+      stateDiff: never;
+    }>
+  | Partial<{
+      code: string;
+      nonce: string;
+      balance: string;
+      state: never;
+      stateDiff: { [slot: string]: string };
+    }>;
+
+export type CallOverrides = {
+  [address: string]: CallOverride;
+};
+
+export type CreateAccessListResult = {
+  accessList: AccessList;
+  gasUsed: Quantity;
+};
+
+/**
+ * Stripped down version of the type from EthereumJs
+ */
+interface VMRunCallOpts {
+  caller?: EthereumJsAddress;
+  data?: Buffer;
+  gasPrice?: BN;
+  gasLimit?: BN;
+  to?: EthereumJsAddress;
+  value?: BN;
+  block?: any;
+}
+
+type CloneVmOptions = {
+  blockchain: Blockchain;
+  common: Common;
+  simulationBlock: Block;
+};
+
+type CloneVmResult = {
+  vm: VM;
+  stateManager: DefaultStateManager;
+  stateTrie: GanacheTrie;
+};
+
+type SimulatorCommonOptions = CloneVmOptions & {
+  options: EthereumInternalOptions;
+  emitStepEvent: boolean;
+  transaction: SimulationTransaction;
+};
+
+type RunCallOptions = SimulatorCommonOptions & {
+  overrides?: CallOverrides;
+};
+
+type CreateAccessListOptions = SimulatorCommonOptions;
+
+type RunCallSetupOptions = {
+  common: Common;
+  stateTrie: GanacheTrie;
+  stateManager: DefaultStateManager;
+  transaction: SimulationTransaction;
+  overrides?: CallOverrides;
+};
+
+type RunCallSetupResult = {
+  runCallOpts: VMRunCallOpts;
+  accessListExclusions: EthereumJsAddress[];
+  addressesOnlyStorage: EthereumJsAddress[];
+  intrinsicGas: bigint;
+};
+
+type InternalRunCallOptions = {
+  blockchain: Blockchain;
+  runCallOpts: VMRunCallOpts;
+  vm: VM;
+  options: EthereumInternalOptions;
+  emitStepEvent: boolean;
+};
+
+const toLightEJSAddress = (address?: Address): EthereumJsAddress => {
+  if (address) {
+    const buf = address.toBuffer();
+    return { buf, equals: (a: { buf: Buffer }) => buf.equals(a.buf) } as any;
+  } else {
+    return null;
+  }
+};
+
+/**
+ * Calls the StateManager's `addWarmedAddress` function for the transaction
+ * caller, all precompiles, and the to address, if applicable.
+ * @param stateManager
+ * @param caller
+ * @param to
+ * @returns An array of precompile addresses
+ */
+const warmDefaults = (
+  stateManager: DefaultStateManager,
+  caller: EthereumJsAddress,
+  to?: EthereumJsAddress
+) => {
+  let precompiles: EthereumJsAddress[] = [];
```

Comment:
Any reason that this isn't a single statement like this?
```suggestion
const precompiles: warmPrecompiles(stateManager);
```

Reply:
I think previously we didn't have the `warmPrecompiles` function and did this in a loop maybe? Definitely way better now, thanks.